sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -1011,6 +1016,139 @@ If a node containing a dynamic parameter is updated through the whole graph
 update API, then any previous updates to the dynamic parameter will be reflected
 in the new graph.
 
+==== Graph-Owned Memory Allocations [[graph-memory-allocations]]
+
+It can be desirable for a graph to own and manage memory allocations for memory
+associated with commands in the graph. This is made possible by using the
+`async_malloc` and `async_free` functions from the
+link:../experimental/sycl_ext_codeplay_async_memory_alloc.asciidoc[sycl_ext_codeplay_async_memory_alloc
+] extension. These commands can be added to a graph either via queue recording
+or explicit graph creation, which will create allocations which are owned and
+managed by that specific `command_graph`.
+
+This can be useful for example when capturing library calls, as the library may
+require some temporary memory for commands they record to the graph. Use of
+these APIs by the library enables the creation and management of these
+allocations within the graph, without the library needing to manage their
+lifetime or be aware that the commands are being recorded to a graph.
```

Comment:
I'm not sure about this, but is this paragraph something that could/should be in the "recording library calls" section of the usage guide?

Reply:
I was also unsure about including this here, I think you're right it could be moved. I think this advice is more for library authors than applications recording library calls though, so I might make a new section for that.

Reply:
Have made this change and expanded it a bit to talk about the allocator use case and providing this as an alternative for graph compatibility.